### PR TITLE
Visit Table for Patient

### PR DESCRIPTION
**Pull request by Johngtka**
Pr features: 
- [X] display the patient visit table when patient choose
- [X] searching possiblity for not choosen patient in empty visit component

ezgif com-video-to-gif

### CODE REVIEW

**Comment by Johngtka** [comment]
S&M commit message proposal: 
`feat: add visit table for select patient and search (#28)`

File: src/app/app.module.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { LayoutModule } from '@angular/cdk/layout';
 import { RouterModule } from '@angular/router';
 
 import { MatIconModule } from '@angular/material/icon';
+import { MatSortModule } from '@angular/material/sort';
```
Comment (macrog) [inline src/app/app.module.ts:11]: not needed

File: src/app/services/store.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { Observable } from 'rxjs';
+
+import { Store } from '../models/store';
+import { enviroment } from 'src/environments/environment.dev';
```
Comment (macrog) [inline src/app/services/store.service.ts:7]: change import path to relative, the same as the above, starting with ../ do not absolute path

File: src/app/visit/visit.component.ts [hunk 3]
```diff
@@ -1,18 +1,45 @@
...
 @Component({
     selector: 'app-visit',
     templateUrl: './visit.component.html',
     styleUrls: ['./visit.component.css'],
 })
-export class VisitComponent {
+export class VisitComponent implements OnInit {
+    constructor(
+        private storeService: StoreService,
+        private snacService: SnackService,
```
Comment (macrog) [inline src/app/visit/visit.component.ts:15]: snac? what is snac? 😄 correct spelling and istall Code Spell Checker extensions

File: src/app/visit/visit.component.ts [hunk 4]
```diff
@@ -1,18 +1,45 @@
...
+    ) {}
     patient!: Patient;
+    store!: Store[];
     ngOnInit(): void {
         this.patient = history.state;
+        this.storeService.getStore().subscribe({
+            next: (data) => (
+                (this.store = data),
+                (this.store = data.sort((a, b) => a.id - b.id))
```
Comment (macrog) [inline src/app/visit/visit.component.ts:24]: `(this.store = data),` remove this part, you are assigning it the value with sort to store in a line below

File: src/assets/i18n/pl.json [hunk 5]
```diff
@@ -15,7 +15,8 @@
         "PATIENT_TABLE_GET_PATIENTS": "Wystąpił błąd podczas próby załadowania pacjentów do tabeli, spróbuj ponownie później",
         "USER_INPUT_DIALOG_CREATE_PATIENT": "Wystapił błąd podczas dodawania nowego pacienta, spróbuj ponownie później",
         "USER_INPUT_DIALOG_UPDATE_PATIENT": "Wystąpił błąd podczas aktualizacji pacjenta, spróbuj ponownie później",
-        "PATIENT_TABLE_DELETE_PATIENT": "Wystąpił błąd podczas usuwania pacjenta, spróbuj ponownie później"
+        "PATIENT_TABLE_DELETE_PATIENT": "Wystąpił błąd podczas usuwania pacjenta, spróbuj ponownie później",
+        "PATIENT_VISIT_CREATE_PATIENT": "Wystąpił błąd podczas tworzenia wizyty dla pacjenta, spróbuj ponownie później"
```
Comment (macrog) [inline src/assets/i18n/pl.json:19]: missing translations in English and Spanish

**Review by macrog — CHANGES_REQUESTED** [review]
Just a few changes

**Review by macrog — APPROVED** [review]
good work ❗

**Pull request merged**